fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java

```diff
@@ -228,9 +234,18 @@ public List<TBrokerFileStatus> loadFileList() {
     public List<List<String>> listFilesAndDirs() {
         List<List<String>> files = Lists.newArrayList();
         try {
+            List<FileStatus> fileStatuses = Lists.newArrayList();
             List<String> pieces = Splitter.on(",").trimResults().omitEmptyStrings().splitToList(path);
             for (String piece : ListUtils.emptyIfNull(pieces)) {
-                List<FileStatus> fileStatuses = HdfsUtil.listFileMeta(piece, new BrokerDesc(properties), false);
+                fileStatuses.addAll(listFilesAndDirs(piece, listRecursively, properties));
+
+                // if the path is directory, return files and sub directories in this directory
+                if (!listRecursively && fileStatuses.size() == 1 && fileStatuses.get(0).isDirectory()) {
```

Comment:
It's better to handle multiple directories.
1. `dirx/f1`, `dirx/d2`. Then `list dir*` get 2 directioris/files.
2. `dir1/f1`, `dir1/d2`, `dir2/f3`, `dir_file_x`. Then `list dir*` gets 4 direcories/files, including the file `dir_file_x`.